quickwit/quickwit-search/src/collector.rs

```diff
@@ -611,16 +630,26 @@ impl Collector for QuickwitCollector {
                     Box::new(collector.for_segment(0, segment_reader)?),
                 ))
             }
-            Some(QuickwitAggregations::TantivyAggregations(aggs)) => Some(
-                AggregationSegmentCollectors::TantivyAggregationSegmentCollector(
-                    AggregationSegmentCollector::from_agg_req_and_reader(
-                        aggs,
-                        segment_reader,
-                        segment_ord,
-                        &self.aggregation_limits,
-                    )?,
-                ),
-            ),
+            Some(QuickwitAggregations::TantivyAggregations(aggs)) => {
+                let mut aggs = aggs.clone();
+                visit_term_aggregations(&mut aggs, &mut |terms_agg| {
+                    // Forward split_size (also shard_size) parameter to segment_size, as
+                    // this is the same in context of Quickwit
+                    if let Some(split_size) = &terms_agg.split_size {
+                        terms_agg.segment_size = Some(*split_size);
```

Comment:
I don't understand. So we have an argument called `split_size` introduced in tantivy even though the parameter makes no sense there.

And it is unused... So you are adding glue code to actually have us write `split_size` into `segment_size`? 

Can't we just remove the `split_size` property in tantivy and populate the `segment_size` thing directly?

Is this a serialization issue?

Reply:
`split_size` which aliases to `shard_size` is supposed to set the number of terms that get send from a node to the root node. That part is not handled in tantivy and therefore unused, but would still make sense in the request structure.
Since we have just one segment in Quickwit currently, we can set it to `segment_size`.

We could alias `shard_size` directly to `segment_size`, but that would kind of incorrect.

Reply:
Can you think of a better solution?

Reply:
We could truncate the results to `split_size` before sending the results between nodes, independently of `segment_size`. But I think we can just forward the parameter for now, until we have more insight how `split_size` and `segment_size` should differ.

Reply:
```
split_size which aliases to shard_size is supposed to set the number of terms that get send from a node to the root node. That part is not handled in tantivy and therefore unused, but would still make sense in the request structure.
Since we have just one segment in Quickwit currently, we can set it to segment_size.
```

That makes no sense at all.

Reply:
Sending data between nodes is not handled in tantivy, it's still obviously useful to control that with a parameter.

No idea why would think that is nonsense.